javascript/lang/best-practice/zlib-async-loop.yaml

```diff
@@ -0,0 +1,19 @@
+rules:
+- id: zlib-async-loop
+  patterns:
+  - pattern-inside: |
+      for (...) {
+          ...
+      }
+  - pattern: zlib.$METHOD(...);
+  - metavariable-regex:
+      metavariable: $METHOD
+      regex: ^.+$(?<!Sync)
+  message: |
+    Creating and using a large number of zlib objects simultaneously
+    can cause significant memory fragmentation.
```

Comment:
What should someone do instead? Stay outside of the loop, or use deflateSync, or something else?

Reply:
According to the docs:

> It is strongly recommended that the results of compression operations be cached to avoid duplication of effort.

I'm not entirely sure what a correct implementation would look like, but I suspect they're saying you should cache the `payload` -> `buffer`, and if you see the same payload again use the cache (memoize). I can add this bit to the message, what do you think? I think the fix will be application-dependent for the most part, e.g. add an `await`, use `deflateSync`, memoize, etc.

Reply:
Gotcha. Would you mind editing the message to how the non-Sync API causes memory fragmentation, and that solutions include `await`, `deflateSync`, or memoizing the result? I think this messaging helps educate developers who end up running across this issue.